### DISCUSSION TITLE
Help for change logo

### DISCUSSION THREAD

**Discussion by Zaisuru** [pglombardo/PasswordPusher#2041] [Q&A]
Hi,

I use pwpush with docker and i have configure the application with my settings.yml

i setting.yml i changed that 
![image](https://github.com/pglombardo/PasswordPusher/assets/32138397/b31fbfbf-a0b6-482e-b92b-c6fb621fd374)

After that i need run the docker with this command
docker run -d       --mount type=bind,source=/root/PasswordPusher/config/settings.yml,target=/opt/PasswordPusher/config/settings.yml -v /var/lib/pwpush/logos:/opt/PasswordPusher/public/logos       -p "5100:5100" pglombardo/pwpush:latest

But i have no change ....

Possible for help please ?

**Comment by pglombardo** [comment]
Hi @Zaisuru,

If you access the logo directly, what happens?

For example:  https://your-pwpusher-url.com/logos/regno.png

**Reply by Zaisuru** [reply]
Hi,
No i don't have this access

http://127.0.0.1/logos/regno.png not found

**Reply by Zaisuru** [reply]
my path 
![image](https://github.com/pglombardo/PasswordPusher/assets/32138397/3b56c43f-46be-4d27-8028-1b9ce952a8b4)

**Reply by pglombardo** [reply]
Are you running on port 80 or port 5100 as shown in your docker command?

**Reply by Zaisuru** [reply]
run with the 5100

**Reply by pglombardo** [reply]
So what does http://127.0.0.1:5100/logos/regno.png give you?

**Reply by Zaisuru** [reply]
Not found

**Reply by pglombardo** [reply]
That's the first issue for sure.  The docker command seems correct.  Shell into the container and browse to `/opt/PasswordPusher/public/logos` - what do you get with an `ls`?

**Reply by Zaisuru** [reply]
i copied the regno.png in /opt/PasswordPusher/public/logos and same error 

![image](https://github.com/pglombardo/PasswordPusher/assets/32138397/c0d734d2-146a-46de-9158-df03543d2a15)

**Reply by pglombardo** [reply]
That's not a docker container so now I'm confused.

**Reply by pglombardo** [reply]
Hey @Zaisuru - I know that this is an issue for you and it can definitely be frustrating for sure.  Let's make sure we are talking about the same thing though - it'll make the whole process go much smoother.

To be more specific on the Docker part:

The application has to be able to access the custom logo to make it a part of the application.  With this there are two options:

1. Put the logo into the container with that docker `-v` option that you posted.  Shell into the container into the `/opt/PasswordPusher/public/logos` directory to assure your logos are there.
2. The alternative is to give a fully qualified URL to `light_logo` such as https://pwpush.fra1.cdn.digitaloceanspaces.com/dev%2Facme-logo.jpg (this is the easier option but you have to host your logos elsewhere)

One more point:

In your first screenshot, you can't put the `PWP__BRAND__LIGHT_LOGO` names into the `settings.yml` - those are shell environment variables for the application.  In the `setting.yml` file, you should be using the `light_logo` and `dark_logo` lines.

I hope this helps a bit.  My advice is to learn what you have to - these skills are useful in all Docker based projects and in IT overall.

Let me know once you sort through the basic issues.  I'm more than happy to help out when/where I can.

**Comment by MindTooth** [comment]
It's even possible to embed the new logo as a image/png (Data URI).  Answer over at [Stack Overflow](https://stackoverflow.com/a/32895908/182868), e.g.:

```
echo "data:image/jpeg;base64,$(base64 DSC_0251.JPG)"
```

Then you store that string as the value for `PWP__BRAND__LIGHT_LOGO`.  That way, you don't have to struggle with mounting a file, etc.
